Enforce Windows 10 requirement for NVDA

josephsl opened the pull request:
### Link to issue number:
Closes #18684 

### Summary of the issue:
NVDA 2025.3 and earlier requires Windows 8.1. With 2026.1 and with Python 3.13 dropping active support for Windows 8.1, NVDA will now require Windows 10 (1507) or later.

### Description of user facing changes:
Notable changes include:

* System requirement set to Widows 10 (in code and in user-facing documentation)
* Windows OneCore will be used as default speech synthesizer.

### Description of developer facing changes:
The following changes were made:

* SConscript: Windows subsystem set to Windows 10
* winVersion: removed recognition of Windows 8.1 (the constant itself is deprecated) and minimum version set to Windows 10
* Removed winVersion.WIN10 checks from the source code, including speech synthesizer list (OneCore is used by default)

### Description of development approach:
Edited the NVDA source code to remove checks on winVersion.WIN10, similar to the approach taken in #15544 and friends.

### Testing strategy:
Manual and tests: Windows 10 is set as minimum required Windows release.

### Known issues with pull request:
Additional work on Windows 10 system requirments will be done in future pull requests, including making Windows OCR available by default.

### Code Review Checklist:

<!--
This checklist is a reminder of things commonly forgotten in a new PR.
Authors, please do a self-review of this pull-request.
Check items to confirm you have thought about the relevance of the item.
Where items are missing (eg unit / system tests), please explain in the PR.
To check an item `- [ ]` becomes `- [x]`, note spacing.
You can also check the checkboxes after the PR is created.
A detailed explanation of this checklist is available here:
https://github.com/nvaccess/nvda/blob/master/projectDocs/dev/githubPullRequestTemplateExplanationAndExamples.md#code-review-checklist
-->

- [x] Documentation:
  - Change log entry
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] UX of all users considered:
  - Speech
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
- [ ] API is compatible with existing add-ons.
- [x] Security precautions taken.
- [ ] NV Access note: When merging, also make changes to the server to prevent those on Win 8.1 from being served updates to 2026.1+

<!-- Please keep the following -->
@coderabbitai summary

### Notes:

* This is an API breaking change in winVersion module where minimum version will return winVersion.WIN10 (and thus add-on API compatibility checkbox is not checked).

josephsl committed:
SConscript: set Windows subsystem version to Windows 10 (for NVDA helper and friends).

josephsl committed:
NVDA helper readme: WINBLUE -> WIN10 (consistency with arch build sconscript)

josephsl committed:
winVersion: minimum Windows version -> Windows 10 (RTM/1507)

josephsl committed:
winVersion: deprecate Windows 8.1 attribute (the attribute will be removed in a future NVDA release)

josephsl committed:
winVersion: remove Windows 8.1 (6.3) detection and attribute and edit comments

josephsl committed:
Synth drivers: set Windows OneCore as default synthesizer and say 'true' when chekcing for synth availability

josephsl committed:
UIA handler: unconditionally add text change and auto-selection detection to evnet handler group map

josephsl committed:
UIA handler: remove text change event check.

Remove Windows 10 check from text change event check.

josephsl committed:
appModules/excel: remove richedit60/Windows version check.

Remove RichEdit60 and Windows version check as Windows 10 or later is required to run NVDA.

josephsl committed:
appModules/outlook: remove richedit60/Windows version check.

Remove RichEdit60 and Windows version check as Windows 10 or later is required to run NVDA.

josephsl committed:
winAPI/session tracking: remove Windows 10 check when determining secure desktop activation.

NVDA runs on Windows 10 and later, therefore remove Widows version check when determining secure desktop activation.

josephsl committed:
User guide: drop Windows 8.1 content (system requirements, eSpeak NG, braille displays, OneCore voices)

josephsl committed:
Update copyright headers

josephsl committed:
Changes: document Windows 10 requirement and winVErsion.WIN81 constant deprecation

pre-commit-ci[bot] committed:
Pre-commit auto-fix

josephsl commented:
Hi,

Actually, Windows OCR is out of scope as it relies on path availability, not Windows release.

Thanks.

josephsl committed:
winVersoin: add type hints.

Review comment from Sena Budd (NV Access): add type hint (returns bool) for isSupportOS. Take this time to add type hints for other functions as well.

josephsl committed:
User guide: update system requirements.

Review comment from Sean Budd (NV Access): starting with NVDA 2026.1, Windows 10 will not be under active support from NVDA, therefore update system requirements as follows:
* Minimum: Windows 10 (no active support). Note the fact that Windows 10 is the last release to support 32-bit systems.
* Recommended: Windows 11 and SErver 2025.

josephsl committed:
What's new: declare Windows 10 (Version 1507) as minimum version under 'important notes' section

LeonarddeR commented on source/synthDriverHandler.py:
```diff
@@ -484,10 +483,8 @@ def getSynthInstance(name, asDefault=False):
-if winVersion.getWinVer() >= winVersion.WIN10:
-	# Default to OneCore on Windows 10 and above
-	defaultSynthPriorityList.insert(0, "oneCore")
+# Default to OneCore on Windows 10 and above
```
Not necessary to keep this
```suggestion
```

LeonarddeR commented on user_docs/en/userGuide.md:
```diff
@@ -42,15 +42,15 @@ Major highlights include:
-  * Note that 32-bit operating systems are no longer under active support.
-  * Note that Windows 8.1 and Windows Server versions older than 2022 are no longer under active support.
+  * Note that 32-bit operating systems are no longer under active support (Windows 10 is the last 32-bit operating system).
+  * Note that Windows 10 (all versions) and Windows Server versions older than 2025 are no longer under active support.
```
Are you sure that server 2022 is no longer actively supported? I think this is a bit premature.

seanbudd replied:
>Are you sure that server 2022 is no longer actively supported? I think this is a bit premature.


>Technically, it is premature to say that Windows Server 2022 is not under active support. But as far as GitHub Actions goes, Server 2025 is the image NV Access uses (Server 2022 is supported until 2031).

When we say "active support" in the user guide we mean support by NV Access, not Microsoft.

seanbudd replied:
Also note mainstream support for 2022 ends 2026/10/13. I think it's likely we will continue some level of support for Server 2022 until mainstream support ends. However, since we're only building and testing on Win 11 / Server 2025 now, we should only recommend that.

josephsl commented:
Hi,

Technically, it is premature to say that Windows Server 2022 is not under active support. But as far as GitHub Actions goes, Server 2025 is the image NV Access uses (Server 2022 is supported until 2031).

Tanks.

josephsl committed:
Synth drivers/OneCore: remove Windows 10 comment as it is unnecessary.

Co-authored-by: Leonard de Ruijter <3049216+LeonarddeR@users.noreply.github.com>

seanbudd commented on user_docs/en/userGuide.md:
```diff
@@ -44,13 +44,13 @@ Major highlights include:
   * both AMD64 and ARM64 variants of Windows are supported, including Copilot+ PCs.
-  * Note that 32-bit operating systems are no longer under active support.
-  * Note that Windows 10 versions earlier than 21H2 and Windows Server versions older than 2022 are no longer under active support.
+  * Note that 32-bit operating systems are no longer under active support (Windows 10 is the last 32-bit operating system).
```
I don't think we should bother updating this line as we will remove it soon, and it's not necessary to clarify in the first place.

```suggestion
  * Note that 32-bit operating systems are no longer under active support.
```

seanbudd commented on user_docs/en/userGuide.md:
```diff
@@ -44,13 +44,13 @@ Major highlights include:
-  * Note that 32-bit operating systems are no longer under active support.
-  * Note that Windows 10 versions earlier than 21H2 and Windows Server versions older than 2022 are no longer under active support.
+  * Note that 32-bit operating systems are no longer under active support (Windows 10 is the last 32-bit operating system).
+  * Note that Windows 10 (all versions) and Windows Server versions older than 2025 are no longer under active support.
```
```suggestion
  * Note that Windows 10 and Windows Server versions older than 2025 are no longer under active support.
```

seanbudd commented on user_docs/en/changes.md:
```diff
@@ -6,7 +6,7 @@
 * This release breaks compatibility with existing add-ons.
 * Windows 8.1 is no longer supported.
-Windows 10 is the minimum Windows version supported.
+Windows 10 (Version 1507) is the minimum Windows version supported.
```
```suggestion
Windows 10 (Version 1507) is the minimum Windows version supported.
We recommend updating to Windows 11, or when that's not possible, to the latest Windows 10 version (22H2).
```

seanbudd commented on source/UIAHandler/__init__.py:
```diff
@@ -250,19 +250,17 @@
+		UIA.UIA_Text_TextSelectionChangedEventId,
+	},
+)
+autoSelectDetectionAvailable = True
```
Let's just remove it in 2026.1, no need to deprecate in 2025.3

seanbudd commented on user_docs/en/userGuide.md:
```diff
@@ -42,15 +42,15 @@ Major highlights include:
 #### Minimum System Requirements {#MinimumSystemRequirements}
 
-* Operating Systems: all 32-bit and 64-bit editions of Windows 8.1, Windows 10, Windows 11, and all Server Operating Systems starting from Windows Server 2012 R2.
+* Operating Systems: all 32-bit and 64-bit editions of Windows 10, Windows 11, and all Server Operating Systems starting from Windows Server 2016.
```
We can relatively confidently update this now

```suggestion
* Operating Systems: 64-bit editions of Windows 10, Windows 11.
Windows Server 2016, 2019, 2022 and 2025.
```

josephsl committed:
What's new: add word on Windows 10 22H2 as recommended Windows 10 versoin (with notes form NV Access)

josephsl committed:
User guide: refinements to system requirements section (with review from NV Access)

josephsl committed:
NVDAObjects.UIA: remove UIAHandler.autoSelectDetectionAvailable attribute check.

Reviewed by Sean Budd (NV Access): remove UIAHandler.autoSelectDetectionAvailable attribute as Windows 10 is the minimum Windows version supported. The only place that checks for this attribute is NVDAObjects.UIA.findOverlayClasses, so remove this attribute check before removing it from UIA handler.

josephsl committed:
UIA handler: remove 'autoSelectDetectionAvailable' attribute as it is no longer used anywhere.

josephsl committed:
What's new: add removal of 'UIAHanlder.autoSelectDetectionAvailable' attribute with no replacement.

josephsl committed:
UIA handler (bonus): remove typing.Optoinal (handler: UIAHandler | None)

seanbudd reviewed (approved):
Thanks @josephsl

seanbudd commented on user_docs/en/userGuide.md:
```diff
@@ -42,10 +42,11 @@ Major highlights include:
   * both AMD64 and ARM64 variants of Windows are supported, including Copilot+ PCs.
-  * Note that 32-bit operating systems are no longer under active support (Windows 10 is the last 32-bit operating system).
-  * Note that Windows 10 (all versions) and Windows Server versions older than 2025 are no longer under active support.
+  * Note that 32-bit operating systems are no longer under active support.
```
I think its safe to remove this like we did for the above change

seanbudd committed:
Update user_docs/en/userGuide.md

josephsl commented:
Hi, if Version 1607 is chosen, please wait a day or two so I can bring in “aqua16” work and edit the documentation accordingly. For Windows 10 22H2, this is required for ESU (extended security updates), and yes on Windows Server 2022 wording. Thanks for reviewing the PR.

josephsl committed:
NVDAObjects.UIA: remove unused import

LeonarddeR commented on user_docs/en/userGuide.md:
```diff
@@ -42,15 +42,15 @@ Major highlights include:
   * both AMD64 and ARM64 variants of Windows are supported, including Copilot+ PCs.
-  * Note that 32-bit operating systems are no longer under active support.
-  * Note that Windows 8.1 and Windows Server versions older than 2022 are no longer under active support.
+  * Note that Windows 10 and Windows Server versions older than 2025 are no longer under active support.
```
I still think the terminology here is ambiguous.
```suggestion
  * Note that Windows 10 and Windows Server versions older than 2025 are no longer under active support by NV Access.
```

seanbudd replied:
Fair, thanks for the suggestion

seanbudd committed:
Update user_docs/en/userGuide.md

Co-authored-by: Leonard de Ruijter <3049216+LeonarddeR@users.noreply.github.com>

LeonarddeR commented:
@seanbudd 

>     * If we want to support server 2022 / win 10 22H2 we should create dual builds with system testing on server-2022

If so, take note of https://github.com/LeonarddeR/nvda/pull/2 . This was a prototype to test windows 11 ARM tests so can easily be expanded to test on server 2022. It's not necessary to reinvent the wheel there 😉

seanbudd commented:
@LeonarddeR - would you consider opening a separate PR for that, similar to #18628, that just adds the matrix changes, with just `windows-2025` in the matrix for now?
I was planning on doing this at some point.

LeonarddeR commented:
I'll do that later today.

josephsl committed:
Merge branch 'master' into win10Requirement

josephsl committed:
Merge branch 'master' into win10Requirement

josephsl committed:
NVDA manifest/compatibility: remove Windows 8.1 manifest content

gerald-hartig commented:
Thanks everyone for the discussion, and great work on this PR @josephsl.
We've reached a decision on the Windows support policy. We will be moving forward with a two-tiered model:

- Minimum requirement: Windows 10 (all versions) and Windows Server 2016.
- Recommended: Windows 11 (Windows Server 2025) and Windows 10 22H2 (Windows Server 2022).

The key reasons for this are:

1. It aligns with our mission to support users for as long as is technically feasible.
2. The "Recommended" tier clarifies for users where our focus is for testing and bug triage.

@josephsl, can you please update this PR to:

- Set the minimum required version in the code to Windows 10 (1507).
- Update the user documentation (user guide, changelog) to reflect the new Minimum and Recommended policy.

Removing the `blocked/needs-product-decision` label.

josephsl committed:
Merge branch 'master' into win10Requirement

josephsl committed:
What's new: tweak Windows version requirement entry to state Windows 10 and 11 (and 10 22H2) are minimum and recommended versions, respectively.

josephsl committed:
User guide: edit Windows release requirements.

* Minimum: Windows 10 (Version 1507), Server 2016 and later
* Recommended: Windows 10 22H2, Windows 11, Server 2022 and 2025

josephsl committed:
Project docs/dev guide: remove 'Windows 8' qualifier on wwahost.exe

josephsl committed:
Project docs/creating dev environment: enforce Windows 10 requireemtn for running and building NVDA from source

josephsl commented:
Hi,

Follow-up task: in addition to Start screen workaround removal (Windows 8.x), can we review the NVDA source code and mark (and remove) code needed for Windows releases prior to Windows 10 please?

Thanks.

seanbudd commented on user_docs/en/changes.md:
```diff
@@ -18,7 +18,9 @@ This can be enabled using the "Report when lists support multiple selection" set
 * NVDA no longer supports Windows 8.1.
-Windows 10 (Version 1507) is the minimum Windows version supported. (#18684, @josephsl)
+Windows 10 (Version 1507) is the minimum Windows version supported.
+We recommend using Windows 11, or if that is not possible, latest Windows 10 release (Version 22H2). (#18684, @josephsl)
```
```suggestion
We recommend using Windows 11, or if that is not possible, the latest Windows 10 release (Version 22H2). (#18684, @josephsl)
```

seanbudd committed:
Apply suggestions from code review

SaschaCowley commented on user_docs/en/userGuide.md:
```diff
@@ -42,15 +42,15 @@ Major highlights include:
 #### Recommended System Requirements {#RecommendedSystemRequirements}
 
-* Operating Systems: 64-bit editions of Windows 10, Windows 11, and Windows Server 2022.
+* Operating Systems: Windows 10 (Version 22H2), Windows 11, and Windows Server 2022 and 2025.
```
```suggestion
* Operating Systems: Windows 11 or Windows 10 (Version 22H2),  and Windows Server 2025 or 2022.
```

josephsl committed:
User guide: reorder Windows 11/11 requireement

Co-authored-by: Sascha Cowley <16543535+SaschaCowley@users.noreply.github.com>

seanbudd committed:
Update user_docs/en/userGuide.md

seanbudd commented:
@josephsl - do you have any thoughts on @Qchristensen latest comment?

josephsl commented:
Hi,

I think using the same wording across what's new and the user guide would be better, although what's new is less verbose and less detailed than the user guide (and keeping the wording the same can hep translators).

Thanks.

seanbudd committed:
Merge branch 'master' into win10Requirement

seanbudd commented on user_docs/en/changes.md:
```diff
@@ -48,9 +52,11 @@ These should be supported natively in Python 3.13. (#18689)
 
 #### Deprecations
 
+* `winVersion.WIN81` constant has been deprecated from winVersion module. (#18684, @josephsl):
```
```suggestion
* `winVersion.WIN81` constant has been deprecated from the `winVersion` module. (#18684, @josephsl):
```

seanbudd committed:
Update user_docs/en/changes.md
